### ISSUE TITLE
impossible to set knex connection options containing underscore for postgres

### ISSUE THREAD

**Issue by danilobuerger** [directus/directus#23241]
### Describe the Bug

Currently its impossible to set postgres db connection options for knex that contain an underscore as all env vars are camelcased:

https://github.com/directus/directus/blob/338625f894c44a6fd505c89743f399b9adb5b0ea/api/src/database/index.ts#L51

https://github.com/directus/directus/blob/338625f894c44a6fd505c89743f399b9adb5b0ea/api/src/utils/get-config-from-env.ts#L8

The affected options are `statement_timeout`, `query_timeout`, `idle_in_transaction_session_timeout`, `application_name`:

https://github.com/knex/knex/blob/176151d8048b2a7feeb89a3d649a5580786d4f4e/types/index.d.ts#L3041-L3061

### To Reproduce

Set `DB_APPLICATION_NAME=meow`. This will get translated to `applicationName: 'meow'` instead of `application_name: 'meow'` and will therefor not be applied. 

### Directus Version

v11.0.0

### Hosting Strategy

Self-Hosted (Custom)

### Database

PostgreSQL 16

**Comment by rijkvanzanten** [comment]
Gotta say, it's absolutely cursed that those options mix camelCase and underscores. Makes it impossible for us to standardize it without maintaining a list of options :C

**Comment by danilobuerger** [comment]
@rijkvanzanten Yes, and I cant offer a good solution. The only thing that comes to mind, which feels dirty, is explicitly transforming those 4 params in `api/src/database/index.ts`. If that is something that is acceptable, I would make a PR.

**Comment by rijkvanzanten** [comment]
Yeah that's the only workaround I can think of as well. My design goal for the API has always been to keep the Directus side as generic as possible, and rely on knex and helpers to translate the individual database parts. We're sprinkling in more and more "if postgres, do XYZ" type of if statements across the codebase.

Nothing to do about that, but a bit unfortunate!

**Comment by paescuj** [comment]
How about using `___` (3 underscores) as a way to escape the camelcasing?

```ini
DB_APPLICATION___NAME=meow
```

**Comment by danilobuerger** [comment]
Apart from the aesthetics (🤢) this makes more sense to me as we don't have to watch for changes in knex and keep a list.

**Comment by paescuj** [comment]
> Apart from the aesthetics (🤢) this makes more sense to me as we don't have to watch for changes in knex and keep a list.

Agreed (🤢😅)

**Comment by rijkvanzanten** [comment]
I also think it'll theoretically conflict with a nested field starting with underscore 🤔 

These would effectively be identical

```
DB_APPLICATION___NAME
{ application: { _name: "x" }}
{ application_name: "x" }
```

Seeing that we parse everything to JSON anyways, maybe we can implement a workaround where the user can set everything in JSON? Something like `DB_OPTIONS="{\"query_timeout\": 1200}"

**Comment by danilobuerger** [comment]
@rijkvanzanten there don't appear to be any starting with an underscore (at least currently)

**Comment by rijkvanzanten** [comment]
@danilobuerger I wouldn't be surprised if within 24h of us shipping a new `___` syntax somebody will show up and be like "HEY I can't use _makeItWork anymore in S3!!" 🫠

**Comment by danilobuerger** [comment]
How about adding a third type to `getConfigFromEnv` so the triple syntax only applies to database envs?

**Comment by danilobuerger** [comment]
So no on the current PR, and rather JSON Options? (If yes, I would rather name it `DB_JSON_CONNECTION` (or similar) so that its clear that its json and applies to the connection options.

**Comment by danilobuerger** [comment]
Ok, well the PR is ready :)

**Comment by rijkvanzanten** [comment]
I think my proposed solution here is to mix JSON values in with the other values, for example:

```
DB_NAME="something"
DB_NESTED_EXAMPLE="{ \"application_name\": \"something else\" }"
```

which should result in

```json
{
  "name": "something",
  "nestedExample": {
    "application_name": "something else"
  }
}
```

While not as elegant as having a key syntax, IMO it beats the triple underscore approach in usability 🙂

**Comment by paescuj** [comment]
> I think my proposed solution here is to mix JSON values in with the other values, for example:
> 
> ```
> DB_NAME="something"
> DB_NESTED_EXAMPLE="{ \"application_name\": \"something else\" }"
> ```
> 
> which should result in
> 
> ```json
> {
>   "name": "something",
>   "nestedExample": {
>     "application_name": "something else"
>   }
> }
> ```
> 
> While not as elegant as having a key syntax, IMO it beats the triple underscore approach in usability 🙂

Sounds good 👍 That means, we have to update the `getConfigFromEnv` function to also consider a root key `DB` or alternatively a connection key `DB_CONNECTION` (if valid JSON):

```ini
DB='{ "client": "sqlite3", "connection": { "filename": "data.db", "application_name": "meow" } }'
# or
DB_CONNECTION='{ "filename": "data.db", "application_name": "meow" }'
```

Without that, the case mentioned in this issue would not be covered, as it is not a nested value.
